arch/arm/src/sama5/hardware/sam_flexcom_spi.h

```diff
@@ -34,10 +34,263 @@
+
+#  define FLEXCOM_SPI_CSR_BITS8           (0 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 8 */
+#  define FLEXCOM_SPI_CSR_BITS9           (1 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 9 */
+#  define FLEXCOM_SPI_CSR_BITS10          (2 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 10 */
+#  define FLEXCOM_SPI_CSR_BITS11          (3 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 11 */
+#  define FLEXCOM_SPI_CSR_BITS12          (4 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 12 */
+#  define FLEXCOM_SPI_CSR_BITS13          (5 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 13 */
+#  define FLEXCOM_SPI_CSR_BITS14          (6 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 14 */
+#  define FLEXCOM_SPI_CSR_BITS15          (7 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 15 */
+#  define FLEXCOM_SPI_CSR_BITS16          (8 << FLEXCOM_SPI_CSR_BITS_SHIFT) /* 16 */
+
+#define FLEXCOM_SPI_CSR_SCBR_SHIFT        (8)       /* Bits 8-15: Serial Clock Baud Rate */
+#define FLEXCOM_SPI_CSR_SCBR_MASK         (0xff << FLEXCOM_SPI_CSR_SCBR_SHIFT)
+#  define FLEXCOM_SPI_CSR_SCBR(n)         ((uint32_t)(n) << FLEXCOM_SPI_CSR_SCBR_SHIFT)
+#define FLEXCOM_SPI_CSR_DLYBS_SHIFT       (16)      /* Bits 16-23: Delay Before SPCK */
+#define FLEXCOM_SPI_CSR_DLYBS_MASK        (0xff << FLEXCOM_SPI_CSR_DLYBS_SHIFT)
+#  define FLEXCOM_SPI_CSR_DLYBS(n)        ((uint32_t)(n) << FLEXCOM_SPI_CSR_DLYBS_SHIFT)
+#define FLEXCOM_SPI_CSR_DLYBCT_SHIFT      (24)      /* Bits 24-31: Delay Between Consecutive Transfers */
+#define FLEXCOM_SPI_CSR_DLYBCT_MASK       (0xff << FLEXCOM_SPI_CSR_DLYBCT_SHIFT)
+#  define FLEXCOM_SPI_CSR_DLYBCT(n)       ((uint32_t)(n) << FLEXCOM_SPI_CSR_DLYBCT_SHIFT)
+
+/* SPI FIFO Mode Register */
+#define FLEXCOM_SPI_FMR_TXRDYM_SHIFT      (1 << 0)  /* Bits 0-1: TX Data Register empty mode */
+#define FLEXCOM_SPI_FMR_TXRDYM_MASK       (3 << FLEXCOM_SPI_FMR_TXRDYM_SHIFT)
+#define FLEXCOM_SPI_FMR_RXRDYM_SHIFT      (1 << 4)  /* Bits 0-1: RX Data Register full mode */
+#define FLEXCOM_SPI_FMR_RXRDYM_MASK       (3 << FLEXCOM_SPI_FMR_RXRDYM_SHIFT)
+#define FLEXCOM_SPI_FMR_TXFTHRES_SHIFT    (1 << 16) /* Bits 16-21: Transmit FIFO threshold */
+#define FLEXCOM_SPI_FMR_TXFTHRES_MASK     (0x3F << FLEXCOM_SPI_FMR_TXFTHRES_SHIFT)
+#define FLEXCOM_SPI_FMR_RXFTHRES_SHIFT    (1 << 24) /* Bits 24-29: Transmit FIFO threshold */
+#define FLEXCOM_SPI_FMR_RXFTHRES_MASK     (0x3F << FLEXCOM_SPI_FMR_RXFTHRES_SHIFT)
+
+/* SPI FIFO Level Register */
+#define FLEXCOM_SPI_FLR_TXFL_SHIFT        (1 << 0)  /* Bits 0-5: Transmit FIFO Level */
+#define FLEXCOM_SPI_FLR_TXFL_MASK         (0x3F << FLEXCOM_SPI_FLR_TXFL_SHIFT)
+#define FLEXCOM_SPI_FLR_RXFL_SHIFT        (16 << 0)  /* Bits 16-21: Transmit FIFO Level */
+#define FLEXCOM_SPI_FLR_RXFL_MASK         (0x3F << FLEXCOM_SPI_FLR_RXFL_SHIFT)
+
+/* SPI Comparison Register */
+#define FLEXCOM_SPI_CMPR_VAL1_SHIFT       (1 << 0)  /* Bits 0-15: First Comparison Value for Received Character */
+#define FLEXCOM_SPI_CMPR_VAL1_MASK        (0xFFFF << FLEXCOM_SPI_CMPR_VAL1_SHIFT)
```

Comment:
please use either upper or lower case for hex constants, but not mix (I think lower is more preferred).

Reply:
Ah, yes - sorry. I actually created this file 18 months ago before I'd paid any attention to the style guide and coding standards. I'll go through the files and check for all of these.